src/controller.py

```diff
@@ -19,99 +18,107 @@ def __init__(self, input_file=None, random_state=None):
         else:
             self.input = HybridInputManager()
 
+        self.game = None
+        self.players = {}
+        self.current_player = None
+
+        self.name_cycle = None
         if random_state is not None:
             random.seed(random_state)
             logging.info(
                 f'Random state is fixed ({random_state})')
 
-    def run(self):
-        logging.info(
-            'Starting game for 4 players.')
+    @property
+    def player_names(self):
+        return self.players.keys()
 
-        self.player_names = names = ['Alpha', 'Bravo', 'Charlie', 'Delta']
-        self.players = players = {name: Player(name) for name in names}
+    def _switch_player(self):
```

Comment:
Is there a meaningful reason to use _-preceded name?
If so, why do we not use _-preceded names in other internal methods like `.end_turn`?

Reply:
My motivation was like that: all methods but _switch_player keep their 'self' object in a consistent state in terms of the game logic. _switch_player, if called several times in a row, would not trigger the infection phase and drawing cards, so it should not be called explicitly. start_game, end_turn and run_single_command can be public from this point of view, because using them in arbitrary order will not break any game rule.